### DISCUSSION TITLE
Unable to recursively init submodules

### DISCUSSION THREAD

**Discussion by Atemu** [magit/magit#5191] [Ideas]
After cloning a repository with submodules, it is my understanding that in order to populate all submodules correctly, you must run `git submodule update --init --recursive`. However, it is not possible to run this command using magit's submodule transient command because the "Update" action does not allow the `--init` flag and the "Populate" action does not allow the `--recursive` flag.

```
Magit 20240320.2336 [>= 3.3.0.50-git], Transient 0.6.0, Git 2.44.0, Emacs 29.3, gnu/linux
```

**Comment by tarsius** [comment]
`magit-submodule-populate` should probably accept the `--recursive` argument. I'll have to think about it and setup a flock of testing repositories.

**Reply by tarsius** [reply]
I've finally gotten around to it. Implemented in fed387282a660debcd57fabf89e1fb6a312797a6.
